src/main/java/com/taboola/rest/api/RestAPIClient.java

```diff
@@ -1,11 +1,13 @@
 package com.taboola.rest.api;
 
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
 import java.util.List;
 import java.util.Objects;
 import java.util.stream.Collectors;
 
+import okhttp3.Interceptor;
```

Comment:
best to avoid coupling to okhttp3 so I prefer creating our own dedicated Interceptor interface